### DISCUSSION TITLE
How to run 100% in serial? How to setup correctly with real test db?

### DISCUSSION THREAD

**Discussion by ScopeSV** [vitest-dev/vitest#5095] [Q&A]
Hi.
I have a project which contains several hundreds of tests.
Many of them do DB queries.

We migrated from jest when vitest was at version 0.34.6. 
Vitest gave us a bit of trouble. It was tough to run stuff in series, so only one test accessed the db simultaneously.
This was no issue in jest.

I ended up running the tests with npx vitest --threads=false --single-thread=true
This seems to work just fine.

I recently discovered the latest version 1.2.2, and that the command line options were removed from the newer releases.
I tried adding 
```
        pool: 'forks',
        poolOptions: {
            threads: {
                singleThread: true,
            },
        },
```
(Got type errors even though I find the properties in the UserConfig interface) 
to the vitest.config.ts and run vitest with npx vitest --config ./vitest.config.ts but it did not work.

Is there a better way? I would hate to migrate back to jest.

**Comment by sheremet-va** [comment]
Did you try `fileParallelism`? https://vitest.dev/config/#fileparallelism-1-1-0

**Reply by ScopeSV** [reply]
Hi! Thanks, it seems to work good!

**Reply by Vindexus** [reply]
Link has been updated to https://vitest.dev/config/#fileparallelism

**Reply by risalfajar** [reply]
I have wasted an hour just searching a solution for this! This is very important for integration testing, and must be stated in the docs, maybe in Debugging or Common Errors section.

**Reply by AriPerkkio** [reply]
> and must be stated in the docs, maybe in Debugging

See https://vitest.dev/guide/debugging.html#debugging. If you have suggestions how to make it even more visible, feel free to open documentation PRs. 

![image](https://github.com/user-attachments/assets/ebc91aa7-7bae-4665-9bec-137f3405dda3)

**Reply by risalfajar** [reply]
I think it's too crucial to be presented as just a Tip, maybe adding a case in the Common Errors section would be perfect.

**Reply by sheremet-va** [reply]
I agree. The https://github.com/vitest-dev/vitest/pull/7975 PR adds a separate guide only for parallelism

**Reply by ramWTG** [reply]
Setting maxConcurrency as 1 in vitest.config can restrict you from running multiple tests in parallel.
